include/seqan3/core/simd/view_to_simd.hpp

```diff
@@ -0,0 +1,719 @@
+        // Hence we can load `simd_traits<simd_t>::max_length` length many elements at once.
+        // Depending on the packing of `simd_t` we can prefetch blocks and store them in the `cached_simd_chunks`.
+        // E.g. assume `simd_t` with length 8 on SSE4 with max length 16.
+        // To fill the 16x16 matrix we need four 8x8 matrices.
+        // Thus, for the 8 sequences we need to load two times 16 consecutive bytes to fill the matrix, i.e. two loads
+        // see figure below.
+        //
+        //       0    1  ...    7 |   8    9  ...   15
+        // 0  [a00, a01, ..., a07]|[a08, a09, ..., a15]  // first load of seq a reads 16 characters
+        // 1  [b00, b01, ..., b07]|[b08, b09, ..., b15]  // first load of seq b reads 16 characters
+        //            ...         |        ...
+        // 7  [g00, g01, ..., g07]|[g08, g09, ..., g15]  // first load of seq g reads 16 characters
+        //    ----------------------------------------
+        // 8  [a16, a17, ..., a23]|[a24, a25, ..., a31]  // second load of seq a reads next 16 characters
+        // 9  [b16, b17, ..., b23]|[b24, b25, ..., b31]  // second load of seq b reads next 16 characters
+        //            ...         |        ...
+        // 15 [g16, g17, ..., g23]|[g24, g25, ..., g31]  // second load of seq g reads next 16 characters
+        //
+        // This quadratic byte matrix can be transposed efficiently with simd instructions.
+        // If the target simd scalar type is bigger we can apply the same mechanism but have then 16 4x4 matrices
+        // (32 bit) or 256 2x2 matrices (64 bit).
+
+        constexpr int8_t max_size = simd_traits<simd_t>::max_length;
+        constexpr int8_t num_chunks = chunks_per_load;
+        std::array<max_simd_type, max_size> matrix{};
+        final_chunk_pos = 0;  // reset the final chunk position, since this could be the last load.
+        // Iterate over each sequence.
+        for (uint8_t i = 0; i < chunk_size; ++i)
+        {  // Iterate over each block depending on the packing of the target simd vector.
+            int8_t last_chunk_pos = -1;
+            for (uint8_t j = 0; j < num_chunks; ++j)
+            {
+                uint8_t pos = j * chunk_size + i; // matrix entry to fill
+                if (cached_sentinel[i] - cached_iter[i] >= max_size) // not in final block
+                {
+                    matrix[pos] = simd::load<max_simd_type>(std::addressof(*cached_iter[i]));
+                    std::advance(cached_iter[i], max_size);
+                    last_chunk_pos += num_chunks; // We have read num_chunks at once.
+                }
+                else  // Loads the final block byte wise in order to not load from uninitialised memory.
+                {
+                    auto & it = cached_iter[i];
+                    max_simd_type & tmp = matrix[pos];
+                    tmp = simd::fill<max_simd_type>(~0);
```

Comment:
why no fill it here with the padding value? and omit the ~0 semantic

Reply:
because the padding value is based on the scalar type of the target vector size which might be bigger than one byte.